gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/MetricContext.java

```diff
@@ -468,6 +468,16 @@ public ContextAwareHistogram contextAwareHistogram(String name) {
     return contextAwareHistogram(name, ContextAwareMetricFactory.DEFAULT_CONTEXT_AWARE_HISTOGRAM_FACTORY);
   }
 
+  /**
+   * Get a {@link ContextAwareHistogram} with a given name.
+   *
+   * @param name name of the {@link ContextAwareHistogram}
+   * @return the {@link ContextAwareHistogram} with the given name
+   */
+  public ContextAwareHistogram contextAwareHistogramWithSlidingTimeWindow(String name, long windowSize, TimeUnit unit) {
```

Comment:
I guess the arguments already speak for "sliding window". We could simply overload function `public public ContextAwareHistogram contextAwareHistogram(..)`. Same below.

Reply:
I discussed with Issac, he preferred this way. If you prefer the other way, I can change it back.

Reply:
Yeah. Shorter function name makes code cleaner. But we should add in the comments about sliding window.